### PR TITLE
Exclude events from WebTools with certain categories

### PR DESCRIPTION
**Pull request by bingzhang**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by sandeep-ps** [comment]
Hi, @bingzhang, I see that there is an existing [code](https://github.com/rokwire/events-manager/blob/develop/utilities/sourceEvents.py#L159) where we check whether the incoming category is known (). I think it may be better to modify that with a continue statement to ignore unknown/excluded categories. That way, we do not have to update included and excluded categories during future changes. Also, please see my comment on this issue: https://github.com/rokwire/events-manager/issues/1085#issuecomment-1574200638. Thanks.

**Comment by bingzhang** [comment]
> Hi, @bingzhang, I see that there is an existing [code](https://github.com/rokwire/events-manager/blob/develop/utilities/sourceEvents.py#L159) where we check whether the incoming category is known. I think it may be better to modify that with a continue statement to ignore unknown/excluded categories. That way, we do not have to update included and excluded categories during future changes. Also, please see my comment on this issue: [#1085 (comment)](https://github.com/rokwire/events-manager/issues/1085#issuecomment-1574200638). Thanks.

update the code. please check again.

File: utilities/sourceEvents.py [hunk 1]
```diff
@@ -128,6 +128,14 @@ def parse(content, gmaps):
         # if not pe.get('location'):
         #     continue
         entry = dict()
+        entry['category'] = pe['eventType'] if 'eventType' in pe else ""
+        if entry['category'] not in eventTypeMap:
+            __logger.warning("find unknown eventType: {}".format(entry['category']))
```
Comment (sandeep-ps) [inline utilities/sourceEvents.py:133]:
```suggestion
            __logger.warning("Found unknown eventType: {}".format(entry['category']))
```

**Review by sandeep-ps — APPROVED** [review]
Logic-wise, this looks good. I have left a minor text suggestion. I briefly tested this using the develop branch Events BB but found no issue. More testing will be needed after the Events BB changes are merged. Thanks.

**Review by sandeep-ps — APPROVED** [review]

**Comment by minump** [comment]
Wouldn't constants.py eventTypeMap need to be changed? I see "Community/Service" and "Ceremony/Service" still in line https://github.com/rokwire/events-manager/blob/ef2a95c21d0fe8de64b0f75cb27f6088980a33b3/utilities/constants.py#L51 
Or maybe I didnt understand the issue correctly?

**Comment by sandeep-ps** [comment]
> Wouldn't constants.py eventTypeMap need to be changed? I see "Community/Service" and "Ceremony/Service" still in line
> 
> https://github.com/rokwire/events-manager/blob/ef2a95c21d0fe8de64b0f75cb27f6088980a33b3/utilities/constants.py#L51
> 
> 
> Or maybe I didnt understand the issue correctly?

I will jump in here. It's removed in a different PR that deals with the mapping. https://github.com/rokwire/events-manager/pull/1088/files. In the issue, I mentioned that the new mapping might just be needed, but we saw that some logic update is needed for the categories to be excluded.

**Pull request merged**